### ISSUE TITLE
Device keeps rebooting after updating to V3.2.35EN

### ISSUE THREAD

**Issue by DadRe-dmt** [GeekMagicClock/smalltv-pro#67]
After updating to V3.2.35EN , my devices keep rebooting after a 5 - 10 seconds.

Luckily i was able to downgrade it to 3.2.31EN, but i guess there is quite a risk, people might brick the device, 
when the device reboots while updating the firmware.

**Comment by GeekMagicClock** [comment]
@DadRe-dmt Please try the updates, thank you.

**Comment by DadRe-dmt** [comment]
Problem is still the same. After a few seconds the device is rebooting.

**Comment by tinju87** [comment]
Are you connected to the internet? Because I face the same problem if the device has Wifi but no internet access.

**Comment by GeekMagicClock** [comment]
> Problem is still the same. After a few seconds the device is rebooting. I tried all new firmware up till 37. The last version that doesn't trigger the reboot, is version 31.
> 
> Problem seems to occur on every single FW update since 31.
> 
> I also notice that after updating any FW higher than 31, the time (clock) stays at 0:00 , it doesn't seem to sync.

OK, I'll check it again, which app do you used when the reboot occurs?

**Comment by GeekMagicClock** [comment]
> Are you connected to the internet? Because I face the same problem if the device has Wifi but no internet access.

Hi, which App has the same problem? I'll try reproduce it.

**Comment by DadRe-dmt** [comment]
> > Problem is still the same. After a few seconds the device is rebooting. I tried all new firmware up till 37. The last version that doesn't trigger the reboot, is version 31.
> > Problem seems to occur on every single FW update since 31.
> > I also notice that after updating any FW higher than 31, the time (clock) stays at 0:00 , it doesn't seem to sync.
> 
> OK, I'll check it again, is it Bitcoin app you used when the reboot occurs? Let's focus on the latest V3.2.37, the time should be updated. Also I have tried bitcoin/stock/weather, but no reboot was observed within a few hours. If that's possible, can you share the settings of the App?

It is not the Bitcoin app in specific. When setting 'autostart' to clock, the reboot is happening as well.
I can make some screenshots, which part of the interface ?

**Comment by tinju87** [comment]
> > Are you connected to the internet? Because I face the same problem if the device has Wifi but no internet access.
> 
> Hi, which App has the same problem? I'll try reproduce it.
> 
> I have tried with my phone's hotspot, close the internet, but haven't found it will reboot, in weather/stock/bitcoin apps.

No app, you just have to boot up and wait. After maximum 2 minutes it crashes and reboots itself this will loop forever. However my default startup app is pictures. I assume that it has to do with having wifi connected but not having internet access because of firewall blocking it.
I have two devices maybe I will fiddle around with the other one.

**Comment by GeekMagicClock** [comment]
> > > Problem is still the same. After a few seconds the device is rebooting. I tried all new firmware up till 37. The last version that doesn't trigger the reboot, is version 31.
> > > Problem seems to occur on every single FW update since 31.
> > > I also notice that after updating any FW higher than 31, the time (clock) stays at 0:00 , it doesn't seem to sync.
> > 
> > 
> > OK, I'll check it again, is it Bitcoin app you used when the reboot occurs? Let's focus on the latest V3.2.37, the time should be updated. Also I have tried bitcoin/stock/weather, but no reboot was observed within a few hours. If that's possible, can you share the settings of the App?
> 
> It is not the Bitcoin app in specific. When setting 'autostart' to clock, the reboot is happening as well. I can make some screenshots, which part of the interface ?

I've tried to set the autostart to "Clock", but seems work all good for a half hour now.

How often does the reboot occurs?
What is the version before the update to 37?
Did you tried with a factory reset?

**Comment by GeekMagicClock** [comment]
> > > Are you connected to the internet? Because I face the same problem if the device has Wifi but no internet access.
> > 
> > 
> > Hi, which App has the same problem? I'll try reproduce it.
> > I have tried with my phone's hotspot, close the internet, but haven't found it will reboot, in weather/stock/bitcoin apps.
> 
> No app, you just have to boot up and wait. After maximum 2 minutes it crashes and reboots itself this will loop forever. However my default startup app is pictures. I assume that it has to do with having wifi connected but not having internet access because of firewall blocking it. I have two devices maybe I will fiddle around with the other one.

OK, does the firewall blocks all tcp/udp connections? Can try close the firewall to check if it the reboot still occurs?

I tested to set the autostart to "Pictures", but seems works too, with no reboot in 10 minutes.
Is it 3.2.37 version?

**Comment by DadRe-dmt** [comment]
> Are you connected to the internet? Because I face the same problem if the device has Wifi but no internet access.

Interesting!  The devices have (strictly filtered) internet connection, BUT........... 
the problem indeed occurs when not allowing specific outgoing connections.
When the device boots, it keeps polling 213.188.196.246 (port 80) which seems part of worldtimeapi.org platform.
This was also the case with firmware 31 and before, but besides polling to worldtimeapi.org it also used
regular NTP 123 for time syncing. Blocking the http connection to worldtimeapi.org did not make the device boot and the time was syncing without a problem.

In my logging i notice that firmwares after 31 don't seem to use NTP protocol for time syncing anymore.
It looks like connecting to worldtimeapi.org during boot became mandatory to sync time ?
If not allowed during boot, the device will enter a bootloop.

When i block outgoing http connections to 213.188.196.246 AFTER the reboot happened, the device remains up and running.
Until i reboot the device again, the problem start happening again.

I rather use trusted NTP servers for unknown devices, than outgoing HTTP connections to some timesync platform. 
This is why i blocked the outgoing HTTP connections to worldtimeapi.org

@GeekMagicClock Hope this helps to resolve the issue.

**Comment by tinju87** [comment]
Interesting findings.
I will not allow IoT devices to use internet access if they do not need to. NTP is SNATed for not bypassing my DNS sinkhole filter but working. Thus uploading to the newest firmware should work. I will test and provide feedback.

**Comment by DadRe-dmt** [comment]
> Interesting findings. I will not allow IoT devices to use internet access if they do not need to. NTP is SNATed for not bypassing my DNS sinkhole filter but working.

 

> Interesting findings. I will not allow IoT devices to use internet access if they do not need to. NTP is SNATed for not bypassing my DNS sinkhole filter but working.

Same here. Very restrictive with especially IoT devices. The majority allows all........ i guess we are not the majority.
I'm blocking any outgoing connection unless required and 'trusted'. 

Since 32 fw , something seems changed. I run local NTP and even with trusted external NTP's , Since 32 FW, whatever NTP server i fill up .........  the device just completely stopped polling 123 NTP during boot. Only worldtimeapi 80 is getting polled over and over again, and seems mandatory for proper boot. After a few denies, the device is booting again.
Seems like the NTP option in the interface is not operational or completely ignored.
I tested it on any of the 9 devices i own, all similar result.

Quite annoying as i miss al the wonderfull updates/improvement on these great little devices.
Because great little device it is! I just love it.
I mainly using it as a crypto ticker, but it's by far the best project in this niche market.

Props to the dev! @GeekMagicClock

**Comment by GeekMagicClock** [comment]
@DadRe-dmt @tinju87 Thank you all, that is quite interest finding. I think I'm one of the majority :). I did have made some time sync strategies to make sure time could be updated correctly, http or udp (NTP), async or sync. It maight be a mistake that I ignored the NTP sync strategy during my debug since 32 and later, but I can't remember the reason.

According to your findings, I think this part need to be reviewed and rewrite again. By default, udp NTP would be the first used to sync time, if that fails(caused by firewall etc by no intent),  http time sync would be used as a backup method to get time correctly.

Very much appreciated for your support.

**Comment by GeekMagicClock** [comment]
[FW_SmallTV-PRO_V3.2.38EN.zip](https://github.com/user-attachments/files/15747737/FW_SmallTV-PRO_V3.2.38EN.zip)

I have restore the UDP NTP time update method, and the worldtimeapi as a http backup.
UDP time sync interval will be 5minutes if the first sync is success, and it will try your ntp server first (if set), but if fails or not set, it will try 5 NTP servers inside["ntp.aliyun.com", "time.windows.com", "pool.ntp.org", "europe.pool.ntp.org", "asia.pool.ntp.org"],  only when the UDP method fails, the http would be used, if the first http sync fails, would try 10 seconds a time, after the first successul http sync, the interval switchs to 5 minutes. But I don't have a firewall to control all the flows, I can't fully reproduce your situation, hope this version will work with luck.

For future updates, may be more http servers would be added as a backup.
Please try if you are convinent, thanks again.

**Comment by DadRe-dmt** [comment]
> [FW_SmallTV-PRO_V3.2.38EN.zip](https://github.com/user-attachments/files/15747737/FW_SmallTV-PRO_V3.2.38EN.zip)
> 
> I have restore the UDP NTP time update method, and the worldtimeapi as a http backup. UDP time sync interval will be 5minutes if the first sync is success, and it will try your ntp server first (if set), but if fails or not set, it will try 5 NTP servers inside["ntp.aliyun.com", "time.windows.com", "pool.ntp.org", "europe.pool.ntp.org", "asia.pool.ntp.org"], only when the UDP method fails, the http would be used, if the first http sync fails, would try 10 seconds a time, after the first successul http sync, the interval switchs to 5 minutes. But I don't have a firewall to control all the flows, I can't fully reproduce your situation, hope this version will work with luck.
> 
> For future updates, may be more http servers would be added as a backup. Please try if you are convinent, thanks again.

Great work sir! 

I have tested the new FW and confirm the bootloop issue has been resolved in V3.2.38EN.
During boot my NTP server is being polled again, time syncing through NTP is working.

Thank you for the quick fix. Much appreciated.

**Comment by DadRe-dmt** [comment]
1 Manually set a NTP server in this case : ntp1.trans-ix.nl

2 Manual reboot/power off the device.

Result : Manually entered NTP server is not being used during boot.
Bootloop starts when during the boot, the device starts polling a load of ntp servers
from various timepools, except the one (server) that is defined manually.(which i allow in my firewall)
My firewall is blocking any non defined connections. 
As soon as the device is unable to connect NTP 123 for a few attempts, the device starts rebooting.

The bootloop can be stopped when after reboot you click the save button again, 
at manual NTP server interface. It immediately polls the manually defined NTP server,
the device succesfully syncs with the defined ntp server and stop rebooting.
It just seems to forget the settings or at least not use it, when the device reboots/ or power off/on.

I'm not sure why the device starts rebooting when certain "NTP 123 / worldtimeapi" connections are blocked or
simply not available.
That behaviour was not happening with FW 31 and before.
Is it a routine ? /  mandatory condition , or is the device actually crashing on those events ?
I will add a screenshot to show what happens during bootloop.

**Comment by DadRe-dmt** [comment]
<img width="957" alt="Knipsel - kopie" src="https://github.com/GeekMagicClock/smalltv-pro/assets/75391786/a35a758d-fd7d-4083-894b-8588ada0fe8d">


Green connections are when i  click "save" at the devices 'time' interface. Bootloop immediately stops. (until manual reboot)

The red connections are blocked connections during boot and result in  bootloop as manually set NTP server is not used.

**Comment by tinju87** [comment]
I still get the reboot loop with FW_SmallTV-PRO_V3.2.38EN.
Just disable the reboot. It is not necessary at all to reboot.

**Comment by DadRe-dmt** [comment]
> I still get the reboot loop with FW_SmallTV-PRO_V3.2.38EN. Just disable the reboot. It is not necessary at all to reboot.

Disabling/stopping the 'reboot' should be the no.1 priority indeed.

**Comment by GeekMagicClock** [comment]
> 1 Manually set a NTP server in this case : ntp1.trans-ix.nl
> 
> 2 Manual reboot/power off the device.
> 
> Result : Manually entered NTP server is not being used during boot. Bootloop starts when during the boot, the device starts polling a load of ntp servers from various timepools, except the one (server) that is defined manually.(which i allow in my firewall) My firewall is blocking any non defined connections. As soon as the device is unable to connect NTP 123 for a few attempts, the device starts rebooting.
> 
> The bootloop can be stopped when after reboot you click the save button again, at manual NTP server interface. It immediately polls the manually defined NTP server, the device succesfully syncs with the defined ntp server and stop rebooting. It just seems to forget the settings or at least not use it, when the device reboots/ or power off/on.
> 
> I'm not sure why the device starts rebooting when certain "NTP 123 / worldtimeapi" connections are blocked or simply not available. That behaviour was not happening with FW 31 and before. Is it a routine ? / mandatory condition , or is the device actually crashing on those events ?
> 
> Basically the problem is similar as before. Whenever http and/or 123 udp connections are blocked/unavailable for a few polls, the device reboots.
> 
> I will add a screenshot to show what connections happen during bootloop. and what happens after re-clicking the save button.



> 1 Manually set a NTP server in this case : ntp1.trans-ix.nl
> 
> 2 Manual reboot/power off the device.
> 
> Result : Manually entered NTP server is not being used during boot. Bootloop starts when during the boot, the device starts polling a load of ntp servers from various timepools, except the one (server) that is defined manually.(which i allow in my firewall) My firewall is blocking any non defined connections. As soon as the device is unable to connect NTP 123 for a few attempts, the device starts rebooting.
> 
> The bootloop can be stopped when after reboot you click the save button again, at manual NTP server interface. It immediately polls the manually defined NTP server, the device succesfully syncs with the defined ntp server and stop rebooting. It just seems to forget the settings or at least not use it, when the device reboots/ or power off/on.
> 
> I'm not sure why the device starts rebooting when certain "NTP 123 / worldtimeapi" connections are blocked or simply not available. That behaviour was not happening with FW 31 and before. Is it a routine ? / mandatory condition , or is the device actually crashing on those events ?
> 
> Basically the problem is similar as before. Whenever http and/or 123 udp connections are blocked/unavailable for a few polls, the device reboots.
> 
> I will add a screenshot to show what connections happen during bootloop. and what happens after re-clicking the save button.

Hi, yes, still need to modify the user defined NTP server, just too late to take effect in the current firmware, or next reboot will take effect.

And the reboot was caused by a bug in the http sync method, I'm working on this, should be soon to have another test version.

Thanks again.

**Comment by DadRe-dmt** [comment]
When looking at the connection flow, it's indeed happening after several failed http polls of worldtimeapi.

During boot routine, first it tries 5 x to connect to a NTP server, then fallback to http syncing. 
After several failed http sync attempts (10 - 20x) ,the bug occurs and the device reboots and repeats the procedure. 
Until it's able to either sync with a NTP server or by http worldtimeapi, then the bootloop stops immediately.

I stated this was not happening in 31 and before, but the problem is also present in earlier FW versions.
I just didn't notice as i was allowing quite a large NTP pool when i was testing.
So the device was able to sync through NTP, just before it tried the fallback http syncing method.  

'Great' to hear it's narrowed down to a bug in http syncing method and only under certain conditions.
And the other small issue of not using user defined NTP during boot, had yet to be fixed.

When people allow all outgoing 123 udp connections and/or http connections, bug will probably almost never occur because they will succesfully sync with a  NTP or with early attempts of HTTP fallback method before the bug occurs.
So the majority will never experience the bug :)

Thanks for all your work and sorry for the confusion i may have caused about the FW versions.

**Comment by GeekMagicClock** [comment]
> When looking at the connection flow, it's indeed happening after several failed http polls of worldtimeapi.
> 
> During boot routine, first it tries 5 x to connect to a NTP server, then fallback to http syncing. After several failed http sync attempts (10 - 20x) ,the bug occurs and the device reboots and repeats the procedure. Until it's able to either sync with a NTP server or by http worldtimeapi, then the bootloop stops immediately.
> 
> I stated this was not happening in 31 and before, but the problem is also present in earlier FW versions. I just didn't notice as i was allowing quite a large NTP pool at that time when testing. So the device was able to sync through NTP, just before it tried the fallback http syncing method.
> 
> 'Great' to hear bootloop is narrowed down to a bug in http syncing method and only under certain conditions. And the other small issue of not using user defined NTP during boot, had yet to be fixed.
> 
> When people allow all outgoing 123 udp connections and/or http connections, bug will probably almost never occur because they will succesfully sync with a NTP or within early attempts of HTTP fallback method before the bug occurs. So the majority will never experience the bug :)
> 
> Thanks for all your work and sorry for the confusion i may have caused about the FW versions.

It's true as your analaisis, you are very good at it, if udp NTP method fails, will try the http method then.
The small issue about user defined NTP server sync problem should be fixed since 3.2.37, but never mind, you could check the 3.2.39, I tested here, it will try the user defined NTP server once saved.

I have made some improvements about the http sync, but I can't block some website in my enviroment, hope you could help test about http sync crash problem.

Thank you very much for your time.

**Comment by GeekMagicClock** [comment]
[Uploading FW_SmallTV-PRO_V3.2.39EN.zip…]()

**Comment by tinju87** [comment]
Looks good for now. No reboot since 20 minutes.
NTP is also working but wrong timezone. I see some weatherapi calls but they are blocked and no further attempts are made when I do not interact with the device itself.

**Comment by DadRe-dmt** [comment]
The bootloop is indeed gone. 5x NTP then It keeps polling worldtimeapi every 3 seconds and endlesly without the device booting.
NTP is working except not the user defined one.

Crypto ticker interface is no longer working although it is polling Binance api.  
Firmware upload interface looks different (older/test FW ?)

**Comment by DadRe-dmt** [comment]
***  Seems like my device is a bit bricked now. I downgraded to 38 and did a reset to factory settings.

What i noticed is that my boot picture is no longer being loaded during boot.
When i go to the 'picture' interface no pictures are visible.
It shows a message : "failed to open dir"
Downgrading / upgrading doesn't seem to fix this particular problem.

* The timezone problem remains when downgrading firmware.

<img width="918" alt="no-pictures - kopie" src="https://github.com/GeekMagicClock/smalltv-pro/assets/75391786/35315b66-b2da-482d-9768-8e6d742d965f">

**Comment by GeekMagicClock** [comment]
> The bootloop is indeed gone. 5x NTP from non defined pool/server, then It keeps polling worldtimeapi endlessly every 3 seconds without the device rebooting. Very happy that the reboot issue has been resolved.
> 
> But....
> 
> NTP not using the user defined one. (during reboot) When you click save in the NTP interface , yes it will poll the defined NTP server once. But when rebooting, it won't poll the user defined NTP. It uses 5 random servers from some of the pools that are defined in the device. Timezone is also incorrect now. It shows Seoul time.
> 
> Crypto ticker interface is no longer working. It polls the binance api but not showing any visual data on the device. BTC 0.00% 1D , no price.
> 
> Firmware upload interface looks different (firmware and filesystem update in plain text) <img alt="fw-update-interface - kopie" width="299" src="https://private-user-images.githubusercontent.com/75391786/338748186-1fc6380a-d390-4e5a-bfc3-88f4c6b33d93.PNG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgxNzkwMjUsIm5iZiI6MTcxODE3ODcyNSwicGF0aCI6Ii83NTM5MTc4Ni8zMzg3NDgxODYtMWZjNjM4MGEtZDM5MC00ZTVhLWJmYzMtODhmNGM2YjMzZDkzLlBORz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjEyVDA3NTIwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZhYmY2MTlkMGJkYzFjZTI2NWViYmE3NDEwMTFlYTUxYTQwYWE1YTI1NGRhY2FhNjgxZDFiOWM4Y2UzNjQyNTgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.oaw_yzMXQTOHCYs_8S5S63WXo6UCMm_ILM47I6Fe5P0">
> 
> 39 FW seems to broke some things but fixed the reboot issue.

Thank you, that's good news about the boot loop.
I have find out some problems after upload the 39 version to you.

1, Bitcoin App [confirmed]. 
2, Check the user defined NTP server during boot.  [ I need to check]
3, Restore the update web page.  [confirmed]
4. If the timezone is changed, maybe some configuration files are broken, you can try set the weather info again. Here on my device seems all right. [ Need you to check]
5. The images in the device seems are all deleted, total:3776KB, almost the same with free space, have you uploaded the filesytem or click the "Clear" button. [ Need you to check]

Gif files could be downloaded [here](https://github.com/GeekMagicClock/gif).

**Comment by DadRe-dmt** [comment]
4. I've tried setting the weather info and noticed that the time switched 12hr notation to 24 hour notation, but timezone remains 
    Korean time no matter what i do.

5.  I tried clear before uploading and also reload doesn't make a difference. 
     I'm unable to upload any png/jpg/gif. Free space remains the same. 'Failed to open dir message' remains the same.
     Guess i'm having some issue with the filesystem on that particular device.

**Comment by GeekMagicClock** [comment]
> 4. I've tried setting the weather info and noticed that the time switched 12hr notation to 24 hour notation, but timezone remains
>    Korean time no matter what i do.
> 5. I tried clear before uploading and also reload doesn't make a difference.
>    I'm unable to upload any png/jpg/gif. Free space remains the same. 'Failed to open dir message' remains the same.
>    Guess i'm having some issue with the filesystem on that particular device.

4/5 the reason are similar, parameters failed to save on the system.
Did you upload the filesystem files on the update page?

**Comment by GeekMagicClock** [comment]
> > > 4. I've tried setting the weather info and noticed that the time switched 12hr notation to 24 hour notation, but timezone remains
> > >    Korean time no matter what i do.
> > > 5. I tried clear before uploading and also reload doesn't make a difference.
> > >    I'm unable to upload any png/jpg/gif. Free space remains the same. 'Failed to open dir message' remains the same.
> > >    Guess i'm having some issue with the filesystem on that particular device.
> > 
> > 
> > 4/5 the reason are similar, parameters failed to save on the system. Did you upload the filesystem files on the update page?
> 
> Not that i am aware of, i uploaded the fw using the 'update firmware' button.
> 
> I noticed this update layout in earlier FW versions, but didn't touch it, as i expect FW should not be uploaded using Filesystem.

OK, can you use chrome browser, and press F12 to enter web debug mode, click the network tab, and refresh the web, and then check if there's any failed links? 
<img width="1212" alt="image" src="https://github.com/GeekMagicClock/smalltv-pro/assets/143304545/fed03b9f-3976-448f-9bde-a343ed2fba80">

**Comment by DadRe-dmt** [comment]
<img width="961" alt="debug - kopie" src="https://github.com/GeekMagicClock/smalltv-pro/assets/75391786/b33b581b-0d0f-46dc-92f7-55eaa3cedd3e">

**Comment by DadRe-dmt** [comment]
When trying to upload a picture (gif) this is the result :
<img width="959" alt="debug2 - kopie" src="https://github.com/GeekMagicClock/smalltv-pro/assets/75391786/9a7c1f45-6464-4918-882d-a44f0972d496">

**Comment by DadRe-dmt** [comment]
https://github.com/GeekMagicClock/smalltv-pro/assets/75391786/a3f658a9-9350-45cc-a9c0-b5b1d9bf1187

**Comment by DadRe-dmt** [comment]
Another of my devices that is working, shows the following :


https://github.com/GeekMagicClock/smalltv-pro/assets/75391786/943bf3f6-f340-4dcf-8af7-435d4e86456a

**Comment by GeekMagicClock** [comment]
Thanks for the videos.
The fail video has input filters, I can't see if other requests failed too, can you help check if the autoplay save failed too ?
I can see the image upload fail, 500 (Internal server error) returned by the device, this caused by fail to open file.

**Comment by DadRe-dmt** [comment]
> Thanks for the videos. The fail video has input filters, I can't see if other requests failed too, can you help check if the autoplay save failed too ? I can see the image upload fail, 500 (Internal server error) returned by the device, this caused by fail to open file.

Autoplay doesn't show any message on the failing device when i click on save.
When i do exactly the same on working device it shows /set autoplay commands

**Comment by GeekMagicClock** [comment]
Yes, can this file be visited? http://your_clock_ip/.sys/city.json, if you change the city in the weather option, this file should be changed, if not, the whole filesystem maybe damaged.

**Comment by DadRe-dmt** [comment]
I can succesfully visit/access that JSON file at /.sys/city.json.

When i change the city code in the weather interface, it does change the city.json file after saving.

But the weather interface showing an 404 error for .sys/key.json when looking at debug mode.

![image](https://github.com/GeekMagicClock/smalltv-pro/assets/75391786/fcbaa2b4-b59d-414c-ac2c-43d81677a16e)

**Comment by DadRe-dmt** [comment]
Maybe it's better to open a new issue for this specific filesystem matter ? 
It seems only 1 of my devices has this problem, It seems to have happened after a lot of upgrading/downgrading actions,
but i can't be completely sure when and how it happened, as i found out by accident.

Ontopic : 

The bootloop issue seems fixed in 39 but the following issues are present.

* Incorrect timezone. (Korean timezone)
* Manual NTP server not being using during boot.
* Bitcoin feature not working anymore.

Today i received a new Block/Clock so i can test from scratch with a fresh device. :)
It currently has firmware .28EN

**Comment by GeekMagicClock** [comment]
> Maybe it's better to open a new issue for this specific filesystem matter ? It seems only 1 of my devices has this problem, It seems to have happened after a lot of upgrading/downgrading actions, but i can't be completely sure when and how it happened, as i found out by accident.
> 
> Ontopic :
> 
> The bootloop issue seems fixed in 39 but the following issues are present.
> 
> * Incorrect timezone. (Korean timezone)
> * Manual NTP server not being using during boot.
> * Bitcoin feature not working anymore.
> 
> Today i received a new Block/Clock so i can test from scratch with a fresh device. :) It currently has firmware .28EN

Yes, it's better to create a new issue to track the filesystem matter.

[FW_SmallTV-PRO_V3.2.40EN.zip](https://github.com/user-attachments/files/15845535/FW_SmallTV-PRO_V3.2.40EN.zip)
Here's change in the new version:
1. Restore the Bitcoin App.
2. Fix Stock App price change percent.
3. Use the user defined NTP server during boot.

 Incorrect time zone -> The timezone will be updated after the weather update, and saved to the filesystem, I can't duplicate this on my device, is the weather info correct ?

**Comment by GeekMagicClock** [comment]
> I can succesfully visit/access that JSON file at /.sys/city.json.
> 
> ![image](https://private-user-images.githubusercontent.com/75391786/339683882-b0abd8c9-c15f-4c90-bc74-4d7c44f4e393.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg0MjE5MjMsIm5iZiI6MTcxODQyMTYyMywicGF0aCI6Ii83NTM5MTc4Ni8zMzk2ODM4ODItYjBhYmQ4YzktYzE1Zi00YzkwLWJjNzQtNGQ3YzQ0ZjRlMzkzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjE1VDAzMjAyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdiZjk5YWFlNjNlM2VmOTIyZTdkMTE1ZDc2MWFkNGI0Mzc5NGM5NDQ1YWNmNTlmM2M4ZWE3ZmNhMTAxZWUwMTgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.63bjsSop_sU6wC5DgXXkg-plUZeupkr6rRfheKndDdo)
> 
> When i change the city code in the weather interface, it does change the city.json file after saving.
> 
> But the weather interface showing an 404 error for .sys/key.json when looking at debug mode.
> 
> ![image](https://private-user-images.githubusercontent.com/75391786/339683203-fcbaa2b4-b59d-414c-ac2c-43d81677a16e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg0MjE5MjMsIm5iZiI6MTcxODQyMTYyMywicGF0aCI6Ii83NTM5MTc4Ni8zMzk2ODMyMDMtZmNiYWEyYjQtYjU5ZC00MTRjLWFjMmMtNDNkODE2NzdhMTZlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjE1VDAzMjAyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVmZjlhOGFjMTJmYWQxYjBlM2I4ODcyYTJhODNmYjRhNWU5ZjYwN2I5ODEzYWJhMDJjZWI4MDUwYjM3NDU4ZWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.nK_YVfWhAan8go8_P8fJBDdd5Sip3kKtBV8_jvtgV1U)

Seems only the folder /image is damaged, can you check if there're gifs under the weather option? It is under the /image/80x80.gif/ folder, may be disappeared ?

**Comment by DadRe-dmt** [comment]
> [FW_SmallTV-PRO_V3.2.40EN.zip](https://github.com/user-attachments/files/15845535/FW_SmallTV-PRO_V3.2.40EN.zip) Here's change in the new version:
> 
> 1. Restore the Bitcoin App.
> 2. Fix Stock App price change percent.
> 3. Use the user defined NTP server during boot.
> 
> Incorrect time zone -> The timezone will be updated after the weather update, and saved to the filesystem, I can't duplicate this on my device, is the weather info correct ?

* 1. __Restore the Bitcoin app__
* 3. __Use the user defined NTP server during boot__
*     __Incorrect time zone__

**3.2.40EN has resolved all issues.**

Boot issue is fixed. (I tried with blocking all connections)

The device is now utilizing the user defined NTP during boot (and uses the devices pools when no manual server is entered.)
When udp 123 is not available the device will fallback to http time sync. It won't crash no more. 

Time syncing issue is resolved. Going to the weather (management) interface and save the city info, did not do anything 
(no connections except polling NTP / HTTP worldtimeapi and if entered the user defined NTP)
But.... when i actually enter the weather info screen on the device it polls openweathermap IP. 
When i allow it (just once) it corrects the timezone. After this action it remembers the timezone and uses NTP / HTTP sync / Manual NTP server to correctly sync the time. When i manually reboot the device, timezone is remembered and time is correctly synced.

I've tested this succesfully on my device that was not properly working. (filesystem issue)
Great work! Very happy with your support. 🥇 

Will open a separate issue for the filesystem issue, but not really a problem if it can't be fixed.
The image map seems to be missing or maybe is corrupted. 

Thanks again for all your hard work!
(i will finally close this issue)
